### DISCUSSION TITLE
Use custom spi pins other than default on arduino mega

### DISCUSSION THREAD

**Discussion by anandspathak** [jgromes/RadioLib#382] [Q&A]
How can I use custom spi pins on mega instead of default ones as these are occupied by lcd screen? Have no idea,
Regards
Anand Pathak
India

**Comment by jgromes** [comment]
If you're referring to Arduino Mega, that only has one SPI bus and cannot change where its pins are located. You can try a software-emulated SPI (e.g. https://github.com/MajenkoLibraries/SoftSPI), then provide that to the constructor, something along the lines of:

```c++
SoftSPI mySPI(miso, mosi, sck);
SX1278 radio = new Module(cs, dio0, rst, dio1, mySPI);
```

Also: This is not an issue - converting to discussion.

Also also: Please don't open issues for things that are not ... well ... issues. That's what discussions are for.

**Reply by anandspathak** [reply]
My apologies for mistake, will make sure that I don't repeat,
Thanks
Best regards,
Anand Pathak

**Comment by anandspathak** [comment]
was busy since your reply, thanks for same, but soft spi configurationfor lora sx1278 didnt work...i mean lora wasnt initialized...
if you have time can u tell me exact pins on mega that can be used as soft spi..???
anyways i have developed a small project using lora sx1278 , intrestingly i havent noticed that reciever has not missed a single packet since last couple of months....i was using someother library before and had lots of misses....but your library has no issues for me so far so good....
thanks, 
regards
ANAND PATHAK

**Reply by jgromes** [reply]
> soft spi configurationfor lora sx1278 didnt work...i mean lora wasnt initialized...

Unfortunately that's not enough information. Can you post the sketch? Was there an error code? What was the debug outptut? Etc. etc.

You should be able to use any digital IO pin for software SPI.

**Reply by anandspathak** [reply]
Actually as of now i am working on other project(due to some urgency) I will revert back when I continue with this soft spi project, hope you help me then too.
Thanks 
Best Regards
Anand Pathak
India
